### PR TITLE
Python 3.11

### LINKED ISSUE #83: Python 3.11 support
If I add py311 to `tox.ini`, i.e.
```ini
[tox]
envlist = py38,py39,py310,py311,black,isort,ssort,pyflakes,pylint,mypy
```
the tox suite fails.

The error boils down to
```
E       NotImplementedError: AST traversal for 'TryStar' is not implemented
```
which is new syntax in python 3.11 and to
```
{'BaseExceptionGroup', 'ExceptionGroup'}
```
not being in `MODULE_BUILTINS`.

### PR DESCRIPTION
**Pull request by cgahr**
add support for python 3.11, closes #83

### CODE REVIEW

**Review by bwhmather — COMMENTED** [review]
Would you be able to add a test to `tests/test_bindings.py`.  I'm pretty sure that it should work, but better to be safe than sorry.

**Comment by cgahr** [comment]
You are absolutely right, I missed that `get_bindings` has to be tested as well.

File: tests/test_bindings.py [hunk 1]
```diff
@@ -1546,3 +1551,84 @@ def test_match_statement_bindings_as():
...
+            excepthandler* handlers,
+            stmt* orelse,
+            stmt* finalbody,
+        )
+    """
+    node = _parse(
+        """
+        try:
+            a = something_stupid()
+        except* (grp_type := ExceptionGroup) as grp:
```
Comment (bwhmather) [inline tests/test_bindings.py:1626]: What madness is this!  I love some of the possibilities enabled by trying to keep syntax uniform.

**Review by bwhmather — DISMISSED** [review]
Looks good.  Thank you very much.

**Comment by bwhmather** [comment]
Pylint unhappy:  https://github.com/bwhmather/ssort/actions/runs/4875780311/jobs/8698742453

**Comment by cgahr** [comment]
> Pylint unhappy: https://github.com/bwhmather/ssort/actions/runs/4875780311/jobs/8698742453

Yes. It's due to `ast.TryStar` not existing. I think the easiest option is upgrading all linters to python 3.11.

File: tests/test_bindings.py (hunk 1 above)
Comment (cgahr) [reply tests/test_bindings.py:1626]: It's really impressive what possibilities there.

But I was honestly surprised that something like
```python
try:
    pass
except* ExceptionGroup as (first, *_):
    pass
```
does not work

**Comment by bwhmather** [comment]
> I think the easiest option is upgrading all linters to python 3.11.

It looks like that's what we've done in the past.  Works for me.

**Review by bwhmather — APPROVED** [review]

**Pull request merged**